src/test/java/sg/dex/starfish/integration/keeper/SquidResolverTest.java

```diff
@@ -0,0 +1,37 @@
+package sg.dex.starfish.integration.keeper;
+
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import sg.dex.starfish.impl.squid.SquidResolverImpl;
+import sg.dex.starfish.util.DID;
+
+import org.web3j.crypto.CipherException;
+import java.io.IOException;
+
+public class SquidResolverTest {
+    private  SquidResolverImpl resolver;
+    private String value1;
+    private String value2;
+    private DID did;
+
+    public SquidResolverTest() throws IOException, CipherException {
+        resolver = SquidResolverImpl.create();
+        value1 = DID.createRandomString();
+        value2 = DID.createRandomString();
+        did = DID.createRandom();
+    }
+
+    @Test
+    public void testRegisterResolve()   {
+        boolean result = resolver.registerDID(did, value1);
+        assertTrue(result);
+        String val = resolver.getDDOString(did);
+        assertEquals(val, value1);
+        result = resolver.registerDID(did, value2);
```

Comment:
Shouldn't this test use a valid DDO for registration?

Reply:
DDO is just **any** string with the length no more than 2048 bytes. Resolver works simply as map with key:value and possibility to update value only by owner (creator of the first key:value). And inside Ethereum there is no way to check whether DDO is valid or not. Before me it was done by Aquarius. Now Aquarius is nowhere used, it seems, it is full responsibility of owner to provide correct DDO. We could provide some starfish side code check (hash or correctness of JSON schema) but it should be cross-platform code and schema and definitely beyond this PL. Moreover Ocean is always changing DDO schema. We discussed it already with @billbsing . It seems we should create our own DDO schema (again, must be cross-platform) and continuously maintain structures and versioning.